ttnctl/cmd/devices_set.go

```diff
@@ -153,13 +153,17 @@ var devicesSetCmd = &cobra.Command{
 			dev.Description = in
 		}
 
-		if in, err := cmd.Flags().GetStringArray("customs-keys"); err == nil && len(in) > 0 {
+		if in, err := cmd.Flags().GetStringArray("custom-attr"); err == nil && len(in) > 0 {
 			for _, v := range in {
 				s := strings.SplitN(v, ":", 2)
-				if dev.CustomKeys == nil {
-					dev.CustomKeys = make(map[string]string, len(in))
+				if dev.Attributes == nil {
+					dev.Attributes = make(map[string]string, len(in))
+				}
+				if len(s) == 2 {
+					dev.Attributes[s[0]] = s[1]
+				} else {
+					dev.Attributes[s[0]] = ""
```

Comment:
Can work with that. 

Then the attributes in ` - - attr-remove ` will made empty. 
So every attributes who arrive with empty values will be deleted. 
But it also mean that with `--attr-add` if there is no value it will delete the attribute.